### PR TITLE
[pre-commit] add ruff, check-sql checks

### PR DESCRIPTION
**Pull request by danking**
We can use this for pyright or mypy but I found both took ~1.5s even on a tiny file like `ci.py`.

The check is slow the first time because it has to install a new venv; however, subsequent executions can use the extant venv.

### CODE REVIEW

File: batch/test/utils.py [hunk 1]
```diff
@@ -1,26 +1,13 @@
...
-def create_batch(client: bc.BatchClient, **kwargs) -> bc.Batch:
-    ...
-
-
-@overload
-def create_batch(client: aiobc.BatchClient, **kwargs) -> aiobc.Batch:
-    ...
-
-
-def create_batch(client: Union[bc.BatchClient, aiobc.BatchClient], **kwargs) -> Union[bc.Batch, aiobc.Batch]:
```
Comment (danking) [inline batch/test/utils.py:23]: I have no idea why mypy hates these types, but it does. It claims the parameter types are the same in both cases.

**Comment by danking** [comment]
@daniel-goldstein curious for your thoughts here; particularly in light of your switch to pyright. We could use this to run pyright automatically.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,5 +1,28 @@
...
+        language: system
+        types: [python]
+        require_serial: true
+      - id: check-auth
+        name: check-auth
+        entry: bash pre-commit-make-shim auth
+        files: auth/
+        language: system
+        types: [python]
+        require_serial: true
```
Comment (daniel-goldstein) [inline .pre-commit-config.yaml:25]: Can we add `gear`, `web_common`, and `hailtop`?

File: Makefile [hunk 3]
```diff
@@ -52,6 +52,13 @@ check-services: $(CHECK_SERVICES_MODULES)
 pylint-%:
 	$(PYTHON) -m pylint --rcfile pylintrc --recursive=y $* --score=n
 
+.PRECIOUS: $(patsubst %, .venvs/%/bin/activate, $(SERVICES))
+.venvs/%/bin/activate: %/pinned-requirements.txt hail/python/dev/pinned-requirements.txt Makefile
+	$(PYTHON) -m venv .venvs/$*
+	source .venvs/$*/bin/activate && \
+	    pip3 install -r $*/pinned-requirements.txt -r hail/python/dev/pinned-requirements.txt && \
+	    pip3 install --editable ./gear ./web_common
```
Comment (daniel-goldstein) [inline Makefile:60]: How come this doesn't need to install `hailtop` as editable?

File: pre-commit-make-shim [hunk 4]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+set -ex
+
+service=$1
+shift
+make pre-commit-check-in-$service-venv FILES="$*"
```
Comment (daniel-goldstein) [inline pre-commit-make-shim:7]: Won't `$*` include `service`? I feel like we don't want that, we want only the files pre-commit is passing.

**Review by daniel-goldstein — CHANGES_REQUESTED** [review]
Can you make optional hooks with `pre-commit`? I feel like because I use `pyright` in-editor, I don't personally need it when committing, and unlike `ruff` it's not a no-brainer because it lags a little.

File: pre-commit-make-shim (hunk 4 above)
Comment (danking) [reply pre-commit-make-shim:7]: The shift drops the service from `$*`

File: Makefile (hunk 3 above)
Comment (danking) [reply Makefile]: Dunno why this didn't cause problems, but I added it.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (danking) [reply .pre-commit-config.yaml:25]: Yes, done.

Comment (danking) [reply .pre-commit-config.yaml:25]: I also added `hail` which, once the venv was set up takes 360ms.

**Comment by danking** [comment]
@daniel-goldstein in the end this was a trivial change.

**Review by daniel-goldstein — APPROVED** [review]
Love to see it

**Pull request merged**